services/local-notifications/core.go

```diff
@@ -101,6 +112,53 @@ func NewService(appDB *sql.DB, network uint64) *Service {
 	}
 }
 
+func (nb *NotificationBody) GetType() NotificationType {
+	return nb.Type
+}
+
+// TODO I suspect that this JsonMarshaller isn't working as expected
+//  RESOLVE : TestTransactionNotification: core_test.go:120:
+//        	Error Trace:	core_test.go:120
+//        	Error:      	Received unexpected error:
+//        	            	Signal was not handled
+//        	Test:       	TestTransactionNotification
+func (n *Notification) MarshalJSON() (body []byte, err error) {
+
+	switch n.Body.GetType() {
+	case TypeTransaction:
+		tx := n.Body.(*NotificationBody)
+		body, err = json.Marshal(tx)
+		if err != nil {
+			return nil, err
+		}
+
+	default:
+		return nil, fmt.Errorf("unknown NotificationType '%s'", n.Body.GetType())
+	}
+
+	alias := struct {
+		ID            common.Hash  `json:"id"`
+		Platform      float32      `json:"platform,omitempty"`
+		Body          []byte       `json:"body"`
```

Comment:
I think the issue is that you declare `Body` as `[]byte`, meaning that it will be `base64` encoded in the `JSON`.

If you want it to serialize as  JSON (i.e `"body": {"something": ...}`) 
I believe you have to use `json.RawMessage`
https://golang.org/pkg/encoding/json/#RawMessage

which is used for fields that are already serialized

Reply:
Yes, this was part of the problem. `json.RawMessage` was useful for delaying json unmarshalling and handling pre-marshalled data.

But the problem I found was because the `Body.Type` had not been set and `signal.send()` was swallowing the error that gave the marshalling failure.